funsor/distributions.py

```diff
@@ -104,7 +110,7 @@ def __getattribute__(self, attr):
         return super().__getattribute__(attr)
 
     @classmethod
-    @functools.lru_cache(maxsize=None)
+    @functools.lru_cache(maxsize=5000)
```

Comment:
Just curious, were you seeing unbounded growth?

Reply:
Not in any existing tests, I just figured it would be good to add a ceiling.